### PR TITLE
Update chip theme

### PR DESCRIPTION
**Pull request by ghsteff**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[x] Other... Please describe: Theme update
```

## Does this PR introduce a breaking change?
```
[x] Yes
[ ] No
```

This updates `ChipTheme`. Existing applications that overwrite or extend `ChipTheme` will have to update to match the new shape

## Other information

Dark mode

https://github.com/reaviz/reablocks/assets/40581813/1875ceec-dfc6-4349-8f57-9599195d41dc

Light mode

https://github.com/reaviz/reablocks/assets/40581813/f6f51b15-aca9-45d9-a8eb-4e63e8602565

Note: The border of the selected default outlined tag in light mode is the same color as the storybook background so you can't see it

### CODE REVIEW

File: src/elements/Chip/ChipTheme.ts [hunk 1]
```diff
@@ -102,69 +138,146 @@ export const chipTheme: ChipTheme = {
...
-      selectable: [
-        baseTheme.colors.default.selectable,
-        'hover:bg-gray-500 hover:border-gray-500'
-      ].join(' '),
-      selected: '!bg-gray-500 text-white'
+      variants: {
+        filled: 'bg-panel border-panel text-panel-content',
+        outline: 'text-panel-content border-panel-content'
+      },
+      selectable: {
```
Comment (amcdnl) [inline src/elements/Chip/ChipTheme.ts:150]: Interesting that selectable changes it this much.

File: tailwind.config.ts [hunk 2]
```diff
@@ -281,26 +277,22 @@ const config: Config = {
...
           active: colorPalette.green[500],
-          hover: colorPalette.green[400],
-          fill: colorPalette.green[100]
+          hover: colorPalette.green[400]
         },
         error: {
           DEFAULT: colorPalette.red[500],
           active: colorPalette.red[500],
-          hover: colorPalette.red[400],
-          fill: colorPalette.red[100]
```
Comment (amcdnl) [inline tailwind.config.ts:291]: Docs need to be updated for this.

File: src/elements/Chip/ChipTheme.ts (hunk 1 above)
Comment (ghsteff) [reply src/elements/Chip/ChipTheme.ts:150]: ye, adds a bunch of hover state styling that's different for the two variants

Comment (amcdnl) [reply src/elements/Chip/ChipTheme.ts:150]: Is there anyway you could use the lighten or darken to achieve this easier? Just an idea

Comment (ghsteff) [reply src/elements/Chip/ChipTheme.ts:150]: Considering sometimes the background color, text color, and border entirely change, I don't think I can simplify it to lightening or darkening 🥲

**Comment by amcdnl** [comment]
You have a conflict now 😝

**Review by amcdnl — APPROVED** [review]

**Pull request merged**